state.go

```diff
@@ -37,19 +37,33 @@ type nextStateInputs struct {
 	FetchedStats   map[raft.ServerID]*ServerStats
 	LeaderID       raft.ServerID
 	IsLeader       bool // this will be true when the server running the autopilot code is the leader
+	CurrentState   *State
+}
+
+func (i *nextStateInputs) getCurrentServerState(id raft.ServerID) (*ServerState, bool) {
```

Comment:
Currently we only need to track the current state to pull server info out. When a server existed in the current state we should carry over its StableSince time and a few other bits of information during the state building process.

This method makes handling of a couple nil pointers simple. Otherwise we would have to have a bunch of nested conditionals down in buildServerState. For the sake of readability of that code, this function exists.